data/StoreRecord.ts

```diff
@@ -147,6 +148,24 @@ export class StoreRecord {
         return this.validator?.validationState ?? 'Unknown';
     }
 
+    get modifiedData(): PlainObject {
+        if (!this.isModified) return {};
```

Comment:
Returning null would be more efficient, if we are potentially collecting this for giant recordsets.  Most records are clean.